driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -692,6 +699,12 @@ else if (fetchSize != Integer.MAX_VALUE)
    * and handle host failover.
    */
   void execute(final RequestHandler.Callback callback, final Statement statement) {
+    if (this.isClosed()) {
+      throw new IllegalStateException("Could not send request, session is closed");
+    }
+    if (cluster.isClosed()) {
+      throw new IllegalStateException("Could not send request, cluster is closed");
+    }
```

Comment:
Closing a cluster also closes all of its sessions automatically, so it would be caught by the first test, we don't need to re-check here.